packages/next/build/webpack/plugins/next-trace-entrypoints-plugin.ts

```diff
@@ -257,15 +257,44 @@ export class TraceEntryPointsPlugin implements webpack.WebpackPluginInstance {
 
             finishModulesSpan.traceChild('get-entries').traceFn(() => {
               compilation.entries.forEach((entry, name) => {
-                if (name?.replace(/\\/g, '/').startsWith('pages/')) {
+                const normalizedName = name?.replace(/\\/g, '/')
+
+                const isPage = normalizedName.startsWith('pages/')
+                const isApp =
+                  this.appDirEnabled && normalizedName.startsWith('app/')
+
+                if (isApp || isPage) {
                   for (const dep of entry.dependencies) {
                     if (!dep) continue
                     const entryMod = getModuleFromDependency(compilation, dep)
 
+                    // since app entries are wrapped in next-app-loader
+                    // we need to pull the original pagePath for
+                    // referencing during tracing
+                    if (isApp && entryMod.request) {
+                      const loaderQueryIdx = entryMod.request.indexOf('?')
+
+                      const loaderQuery = new URLSearchParams(
```

Comment:
Should we use `new URL(entryMod.request).searchParams` so we dont need to handle the case when there is no query string? 

If `?` is missing, then this seems wrong because it effectively becomes `URLSearchParams(entryMod.request)`

Reply:
If the `?` is missing then something is pretty broken in this field as it needs to follow webpack's loader syntax which passes options via the query.